### PR TITLE
fix(ripple): Fallback to use CSS only ripple in Edge.

### LINKED ISSUE #663: Edge: JS Button problem
When using/testing mdc-web on the demo site, there seems to be an issue with the js button.

### What MDC-Web Version are you using?

0.11.1 via this page:
https://material-components-web.appspot.com/

### What browser(s) is this bug affecting?

Microsoft Edge 41.16193.1001.0

### What OS are you using?

Windows 10 Build 16193

### What are the steps to reproduce the bug?

Open Edge to https://material-components-web.appspot.com/button.html and click on a button that is not CSS only. 

### What is the expected behavior?

It behaves normally or  degrades gracefully to a CSS only button (which appears to be what happens on IE 11)

### What is the actual behavior?
js-button-edge

### PR DESCRIPTION
**Pull request by yeelan0319**
Closes #663.

### CODE REVIEW

File: packages/mdc-ripple/util.js [hunk 1]
```diff
@@ -34,7 +34,16 @@ export function supportsCssVariables(windowObj) {
     windowObj.CSS.supports('(--css-vars: yes)') &&
     windowObj.CSS.supports('color', '#00000000')
   );
-  return explicitlySupportsCssVars || weAreFeatureDetectingSafari10plus;
+  // See: https://caniuse.com/#search=custom%20properties
+  // See: README section on Edge
+  const weAreFeatureDetectingEdge = (
```
Comment (amsheehan) [inline packages/mdc-ripple/util.js]: This should be named something less human. It's full on uncanny valley.

`supportsMSIMEAlignAuto` maybe

Comment (yeelan0319) [inline packages/mdc-ripple/util.js]: But won't it make it less clear that our goal is FD Edge? So when we change the FD feature, we are not always changing the function name?

**Comment by yeelan0319** [comment]
@kfranqueiro I think you raised an very interesting solution and I like it!

Another worry I have with this approach is a more specific FD definitely will lower the "false positive" rate (imagine there is a Edge browser that have CSS Variables correctly implemented a day and in my method it raises the false alarm) but at the same time increase the "false negative" rate (imagine another day that a different Edge version comes out with CSS Variables not correct but not applied to its parent as well, in this case it will think it is right). So there are pros and cons to both method.

I think if we want to be more specific, we need further to valid:
- Animation is somehow tested as well
- The behavior of applying style to its parent is consistent across all Edge version

WDYT?

**Comment by yeelan0319** [comment]
@kfranqueiro @amsheehan I am not very familiar with the cons of UA sniffing besides it sounds to me a little bit unreliable?

**Comment by kfranqueiro** [comment]
I think typically the encouragement towards feature detection over UA sniffing is because user agents can be easily spoofed and even vendors themselves have been known to do funny things to them to defeat developers' workarounds, so it becomes a never-ending battle.  In this particular case though, the efficacy of the feature detect is also in question.

I'm currently downloading the preview release VM to see if its version of Edge behaves the same, and I'll see what versions I have on my own machines tonight.

**Comment by kfranqueiro** [comment]
I was able to test with Edge 15.15063 and 15.16215 (via modern.ie VMs) which both exhibit the same behavior which my POC detects.  I was able to test Edge 14.14393 on my personal machine which does not report support of custom properties at all, so this problem is specific to Edge 15 (14 already won't upgrade ripples).

**Comment by yeelan0319** [comment]
Scoped FD sounds like a plan! For the moment have the good willing that Edge will make it behave as other browsers do.

@kfranqueiro I will close this PR and let you file a new one with scoped FD then 👍

**Pull request closed without merging**